src/Mvc/Mvc.ApiExplorer/src/DefaultApiDescriptionProvider.cs

```diff
@@ -658,12 +658,32 @@ public void WalkParameter(ApiParameterDescriptionContext context)
                 ModelMetadata = bindingContext.ModelMetadata,
                 Name = GetName(containerName, bindingContext),
                 Source = source,
-                Type = bindingContext.ModelMetadata.ModelType,
+                Type = GetModelType(bindingContext.ModelMetadata),
                 ParameterDescriptor = Parameter,
                 BindingInfo = bindingContext.BindingInfo
             };
         }
 
+        private static Type GetModelType(ModelMetadata metadata)
+        {
+            // IsParseableType || IsConvertibleType
+            if (!metadata.IsComplexType)
+            {
+                return metadata.UnderlyingOrModelType.IsPrimitive
+                    // Those additional types have TypeConverter or TryParse and are not primitives
+                    // but should not be considered string in the metadata
+                    || metadata.UnderlyingOrModelType == typeof(DateTime)
+                    || metadata.UnderlyingOrModelType == typeof(DateTimeOffset)
+                    || metadata.UnderlyingOrModelType == typeof(DateOnly)
+                    || metadata.UnderlyingOrModelType == typeof(TimeOnly)
+                    || metadata.UnderlyingOrModelType == typeof(decimal)
+                    || metadata.UnderlyingOrModelType == typeof(Guid)
+                    || metadata.UnderlyingOrModelType == typeof(Uri) ? metadata.ModelType : typeof(string);
```

Comment:
Where did this list of parameter types come from? `EndpointMetadataApiDescriptionProvider` currently translates these types to `string`, right?

https://github.com/dotnet/aspnetcore/blob/9b154d508114a26b45c606dfac35ffc2a0e4dfee/src/Mvc/Mvc.ApiExplorer/src/EndpointMetadataApiDescriptionProvider.cs#L288-L292

Does this difference in ApiExplorer output reflect a difference behavior between ApiController actions and minimal route handlers that take these parameter types?

Reply:
> Where did this list of parameter types come from?

It is a combination of https://swagger.io/specification/ and https://github.com/domaindrivendev/Swashbuckle.AspNetCore/blob/master/src/Swashbuckle.AspNetCore.SwaggerGen/SchemaGenerator/JsonSerializerDataContractResolver.cs#L224

Reply:
> `EndpointMetadataApiDescriptionProvider` currently translates these types to `string`, right?
> Does this difference in ApiExplorer output reflect a difference behavior between ApiController actions and minimal route handlers that take these parameter types?

Exactly, and because of this we have issues like this https://github.com/dotnet/aspnetcore/issues/39886